### ISSUE TITLE
Save custom metric in "--save_epoch_freq".

### ISSUE THREAD

**Issue by houzeyu2683** [junyanz/pytorch-CycleGAN-and-pix2pix#921]
I read the resource about metric of image segmentation.
I use the `pix2pix` model to do my task.
The input is an input and target like blow.
<img src="https://user-images.githubusercontent.com/42731603/74712281-e628e800-5260-11ea-8e2c-ca1bf097dfe8.png" alt="drawing" width="120"/>
<img src="https://user-images.githubusercontent.com/42731603/74712297-ee812300-5260-11ea-802d-5dc68d77f373.png" alt="drawing" width="120"/>

And I already get the prediction result following `!python test.py` and get the result.
The prediction is look like blow.
<img src="https://user-images.githubusercontent.com/42731603/74712328-ffca2f80-5260-11ea-9553-1d8597552118.png" alt="drawing" width="120"/>

I define the `IoU_mean` for evaluation my test data.
```
import numpy as np
from sklearn.metrics import jaccard_similarity_score
def IoU_mean(real, pred):
    real_x = (real>0).astype('int')
    fake_x = (pred>0).astype('int')
    real_r, real_g, real_b = real_x[:,:,0], real_x[:,:,1], real_x[:,:,2]
    fake_r, fake_g, fake_b = fake_x[:,:,0], fake_x[:,:,1], fake_x[:,:,2]
    score_r = jaccard_similarity_score(real_r, fake_r)
    score_g = jaccard_similarity_score(real_g, fake_g)
    score_b = jaccard_similarity_score(real_b, fake_b)
    score_mean = np.mean([score_r, score_g, score_b])
    return(score_mean)
```

How do I save the history of my custom metric to the "IoU.txt" when run the `!python test.py` command?

Thanks.
